client/src/main/java/io/pravega/client/stream/impl/Pinger.java

```diff
@@ -40,6 +47,9 @@
     private ScheduledExecutorService executor = ExecutorServiceHelpers.newScheduledThreadPool(1,
             "pingTxnThread");
     private final List<UUID> txnList = Collections.synchronizedList(new ArrayList<>());
+    @Getter(value = AccessLevel.PACKAGE)
```

Comment:
This is about the line above (47):
1. Why do you need a dedicated thread pool? Each execution of this class' methods is pretty quick so we are wasting a thread for nothing.
2. You have a constructor that takes in an external executor, yet in `close` you shut it down. You shouldn't do that as someone else may be using it.

If all you need an Executor for is to be able to stop the pinger, then `scheduleAtFixedRate` returns a `ScheduledFuture` which can be cancelled when you close this class.

Reply:
Yes, there is no need of an extra thread pool. 
Refactored the implementation to reuse the internal executor provided by `io.pravega.client.netty.impl.ConnectionFactory#getInternalExecutor`